PSCompatibilityCollector/Microsoft.PowerShell.CrossCompatibility/Common/ReadOnlySet.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.PowerShell.CrossCompatibility
```

Comment:
I wonder if this has enough utility to be part of PowerShell (or .NET)

Reply:
I really should be somewhere, I've wanted it a few times